src/client/Client.cc

```diff
@@ -9767,6 +9767,8 @@ void Client::getcwd(string& dir, const UserPerm& perms)
 int Client::statfs(const char *path, struct statvfs *stbuf,
 		   const UserPerm& perms)
 {
+  unsigned long int total_files_on_fs = get_root()->rstat.rfiles +
+    get_root()->rstat.rsubdirs;
   Mutex::Locker l(client_lock);
```

Comment:
Access to client state (such as the root inode's rstats) needs to be after taking client_lock, rather than before.

Reply:
I did the same thing earlier but the ceph-fuse crashes that way -

```
$ ps -e | grep ceph && df -i && ps -e | grep ceph
 9000 ?        00:00:00 ceph-mon
 9212 ?        00:00:02 ceph-mgr
 9656 ?        00:00:01 ceph-osd
 9973 ?        00:00:01 ceph-osd
10289 ?        00:00:00 ceph-osd
10440 pts/0    00:00:00 ceph-mds
10608 pts/0    00:00:00 ceph-fuse
df: ‘/mnt/cephfs’: Software caused connection abort
Filesystem       Inodes  IUsed    IFree IUse% Mounted on
/dev/vda1      83885504 984190 82901314    2% /
devtmpfs        2028238    346  2027892    1% /dev
tmpfs           2033398      1  2033397    1% /dev/shm
tmpfs           2033398    452  2032946    1% /run
tmpfs           2033398     16  2033382    1% /sys/fs/cgroup
tmpfs           2033398      1  2033397    1% /run/user/1000

$ ps -e | grep ceph
 9000 ?        00:00:00 ceph-mon
 9212 ?        00:00:02 ceph-mgr
 9656 ?        00:00:01 ceph-osd
 9973 ?        00:00:01 ceph-osd
10289 ?        00:00:00 ceph-osd
10440 pts/0    00:00:00 ceph-mds


$ df -i
df: ‘/mnt/cephfs’: Transport endpoint is not connected
Filesystem       Inodes  IUsed    IFree IUse% Mounted on
/dev/vda1      83885504 984193 82901311    2% /
devtmpfs        2028238    346  2027892    1% /dev
tmpfs           2033398      1  2033397    1% /dev/shm
tmpfs           2033398    452  2032946    1% /run
tmpfs           2033398     16  2033382    1% /sys/fs/cgroup
tmpfs           2033398      1  2033397    1% /run/user/1000
$ echo $?
1
$
```

Reply:
That's surprising.  You'd need to investigate that crash -- we can't just access things outside of the lock..

Reply:
Resolved. My fix needed a fix. :P